toot/commands.py

```diff
@@ -142,19 +143,54 @@ def _get_scheduled_at(scheduled_at, scheduled_in):
 
 
 def _upload_media(app, user, args):
-    # Match media to corresponding description and upload
+    # Match media to corresponding description and thumbnail
     media = args.media or []
     descriptions = args.description or []
+    thumbnails = args.thumbnail or []
     uploaded_media = []
 
     for idx, file in enumerate(media):
         description = descriptions[idx].strip() if idx < len(descriptions) else None
-        result = _do_upload(app, user, file, description)
+        thumbnail = thumbnails[idx] if idx < len(thumbnails) else None
+        result = _do_upload(app, user, file, description, thumbnail)
         uploaded_media.append(result)
 
+    _wait_until_all_processed(app, user, uploaded_media)
+
     return [m["id"] for m in uploaded_media]
 
 
+def _wait_until_all_processed(app, user, uploaded_media):
+    """
+    Media is uploaded asynchronously, and cannot be attached until the server
+    has finished processing it. This function waits for that to happen.
+
+    Once media is processed, it will have the URL populated.
+    """
+    if all(m["url"] for m in uploaded_media):
+        return
+
+    # Timeout after waiting 1 minute
+    start_time = time()
+    timeout = 60
```

Comment:
This is a troublesome Mastodon API; you are forced to poll indefinitely and have no idea how long the server will need to process the media. 60 seconds is reasonable, but to account for large files or slow servers, should this be (yet another) config option?

Reply:
Yeah, it would be good to be able to customize this. Once I finally add a settings file... Although i think 1 minute should be plenty in the vast majority of cases.